examples/gitops/argocd-codecommit/README.md

```diff
@@ -0,0 +1,169 @@
+# EKS Cluster with ArgoCD and Workloads in private AWS CodeCommit repository
+
+This example shows how to provision an EKS cluster with:
+
+- ArgoCD
+  - Workloads and addons deployed by ArgoCD (main.ft, locals.tf, providers.tf, outputs.tf, data.tf)
+  - Private AWS CodeCommit repository and AWS Lambda stack provisioned by included Terraform script (repo-cc.tf) and used for workloads
+  - AWS Lambda triggers ArgoCD projects sync using ArgoCD webhook after each push to the AWS CodeCommit repository
+
+![Architectural diagramm](images/argocd-cc.png)
+
+To better understand how ArgoCD works with EKS Blueprints, read the EKS Blueprints ArgoCD [Documentation](https://aws-ia.github.io/terraform-aws-eks-blueprints/latest/add-ons/argocd/)
+
+## Reference Documentation
+
+- [Documentation](https://aws-ia.github.io/terraform-aws-eks-blueprints/latest/add-ons/argocd/)
+- [EKS Blueprints Add-ons Repo](https://github.com/aws-samples/eks-blueprints-add-ons)
+- [EKS Blueprints Workloads Repo](https://github.com/aws-samples/eks-blueprints-workloads)
+- [EKS Blueprints for Terraform Workshop](https://catalog.workshops.aws/eks-blueprints-terraform/en-US)
+
+## Prerequisites
+
+Main Terraform files (main.ft, locals.tf, providers.tf, outputs.tf, data.tf) were created using instructions from [EKS Blueprints for Terraform Workshop](https://catalog.workshops.aws/eks-blueprints-terraform/en-US)
+Solution specifc sript is repo-cc.tf
```

Comment:
Specific

Reply:
Argo is triggered by webhook. Github does it out of the box and for Codecommit additional setup is required: trigger and lambda to call ArgoCD webhook url.
The solution will also work without trigger and webhook, but it this case Argo will use default periodic (every X minutes) calls to CodeCommit repo to check if something changed.
Trigger and Lambda is an example of alternative approach which can speed up feedback for developer and remove unnecessary sync calls.

Reply:
> Specific

fixed